README.md

```diff
@@ -223,9 +223,10 @@ Gitleaks offers a configuration format you can follow to write your own secret d
 title = "Gitleaks title"
 
 # Extend the base (this) configuration. When you extend a configuration
-# the base rules take precedence over the extended rules. I.e., if there are
-# duplicate rules in both the base configuration and the extended configuration
-# the base rules will override the extended rules.
+# the extended rules take precedence over the base rules. I.e., if there are
```

Comment:
This isn't strictly true, as #1556 explicitly causes our config to overwrite values in the extended config.

I think we need better language to describe the relationship. "Base" is used ambiguously in a few places, e.g., in the code "base" refers to the extended rule [here](https://github.com/gitleaks/gitleaks/blob/8ea608595ce1c992166d710b217f59965d7a44d7/config/config.go#L244).

Reply:
Yes, you are right. I somehow struggle a bit to suggest a proper text here. @rgmz Do you have a good suggestion?

Reply:
@rgmz

- "custom" would be a better word for the user-defined config. "base" can cause confusion, as it maybe misunderstood to refer the default config (gitleaks provided).
- Users would generally want "custom" user-defined config rules to take precedence (override) over the "inherited" rules from the extended / parent / default config.
- If there are exceptions on this overriding, it maybe noted.

I hope that these word suggestions help to form a better text here. Let me know if I can help.

Reply:
@zricethezav @rgmz @amithm7 I rewrote the README.md regarding extending rules, please take a look.